src/js/views/maps/ViewFinderView.js

```diff
@@ -0,0 +1,155 @@
+    },
+
+    /** 
+     * Values meant to be used by the rendered HTML template.
+     */
+    templateVars: {
+      errorMessage: "",
+      inputValue: "",
+      placeholder: "Search by latitude and longitude",
+      classNames,
+    },
+
+    /**
+     * @typedef {Object} ViewFinderViewOptions
+     * @property {Map} The Map model associated with this view allowing control
+     * of panning to different locations on the map. 
+     */
+    initialize(options) {
+      this.mapModel = options.model;
+    },
+
+    render() {
+      this.focusInput();
+
+      this.el.innerHTML = _.template(Template)(this.templateVars);
+    },
+
+    /**
+     * Focus the input field on a delay to allow for the input to be
+     * visible on the page before attempting to focus.
+     */
+    focusInput() {
+      _.defer(() => {
+        const input = this.getInput();
+        input.focus();
+        // Move cursor to end of input.
+        input.val("");
+        input.val(this.templateVars.inputValue);
+      });
+    },
+
+    getInput() {
+      return this.$el.find(`.${classNames.input}`);
+    },
+
+    getButton() {
+      return this.$el.find(`.${classNames.button}`);
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    keyup(event) {
+      this.templateVars.inputValue = this.getInput().val();
+      if (event.key === "Enter") {
+        this.search();
+      }
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    search() {
+      this.clearError();
+
+      const coords = this.parseValue(this.templateVars.inputValue)
+      if (!coords) return;
+
+      this.model.zoomTo({ ...coords, height: 321321 /* meters */ });
+    },
+
+    /**
+     * Parse the user's input as a pair of floating point numbers. Log errors to the UI
+     * @return {{Number,Number}|undefined} Undefined represents an irrecoverable user input,
+     *   otherwise returns a latitude, longitude pair.
+     */
+    parseValue(value) {
+      const matches = value.match(floatsRegex);
+      const hasBannedChars = value.match(bannedCharactersRegex) != null;
+      if (matches?.length !== 2 || isNaN(matches[0]) || isNaN(matches[1]) || hasBannedChars) {
+        this.setError("Try entering a search query with two numerical values representing a latitude and longitude (e.g. 64.84, -147.72).");
+        return;
+      }
+
+      const latitude = Number(matches[0]);
+      const longitude = Number(matches[1]);
+      if (latitude > 90 || latitude < -90) {
+        this.setError("Latitude values outside of the range of -90 to 90 may behave unexpectedly.");
+      } else if (longitude > 180 || longitude < -180) {
+        this.setError("Longitude values outside of the range of -180 to 180 may behave unexpectedly.");
+      }
+
+      return { latitude, longitude };
+    },
+
+    clearError() {
+      this.setError("");
+    },
+
+    setError(errorMessage) {
+      this.templateVars.errorMessage = errorMessage;
+      this.render();
```

Comment:
are we calling render() here because setting templateVars doesn't update the html automatically?

Reply:
Correct, there's no concept of bound variables as far as I can tell. Typically you'd register the render function as an event listener on a Model's change event. However, this templateVars object is not a Model, and I don't necessarily want to re-render every time it changes anyhow (like when I'm tracking the input's value)